<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>app</class>
 <widget class="QMainWindow" name="app">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1303</width>
    <height>767</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>OS-master-常琪凯</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="button_scheduling">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>640</y>
      <width>110</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>调度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_ready">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>10</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>500</width>
      <height>500</height>
     </size>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>127</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>63</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Dark">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>127</green>
          <blue>127</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Mid">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>170</green>
          <blue>170</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Shadow">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>127</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>240</red>
          <green>240</green>
          <blue>240</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>255</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>227</red>
          <green>227</green>
          <blue>227</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Dark">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>160</red>
          <green>160</green>
          <blue>160</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Mid">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>160</red>
          <green>160</green>
          <blue>160</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>240</red>
          <green>240</green>
          <blue>240</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Shadow">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>105</red>
          <green>105</green>
          <blue>105</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>245</red>
          <green>245</green>
          <blue>245</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Light">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>127</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Midlight">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>63</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Dark">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>127</green>
          <blue>127</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Mid">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>170</green>
          <blue>170</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>255</green>
          <blue>255</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Shadow">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="AlternateBase">
        <brush brushstyle="SolidPattern">
         <color alpha="255">
          <red>245</red>
          <green>245</green>
          <blue>245</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>活动就绪</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_run">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>111</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>运行队列</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_run">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>640</y>
      <width>110</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>运行</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_end">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>10</y>
      <width>181</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>结束队列</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_end_2">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>10</y>
      <width>201</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>内存分配情况</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="input_pid">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>330</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="input_space">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>370</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="input_runtime">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>410</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>330</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>进程名称</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>370</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>所需空间</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>410</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>运行时间</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>490</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>进程属性</string>
    </property>
   </widget>
   <widget class="QComboBox" name="input_attribute">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>490</y>
      <width>97</width>
      <height>28</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>独立进程</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>并发进程</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>530</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>前驱节点</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_addProgress">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>580</y>
      <width>104</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>添加进程</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>450</y>
      <width>77</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>优先级：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="input_priority">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>450</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="input_preprogress">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>530</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QTableView" name="tableBack">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>330</y>
      <width>380</width>
      <height>234</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>380</width>
      <height>255</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="label_back">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>290</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>后备队列</string>
    </property>
    <property name="textFormat">
     <enum>Qt::MarkdownText</enum>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTableView" name="tableReady">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>380</width>
      <height>241</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>380</width>
      <height>255</height>
     </size>
    </property>
   </widget>
   <widget class="QTableView" name="tableRun">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>50</y>
      <width>295</width>
      <height>171</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>295</width>
      <height>400</height>
     </size>
    </property>
   </widget>
   <widget class="QTableView" name="tablesuspend">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>260</y>
      <width>295</width>
      <height>131</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>295</width>
      <height>400</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="label_suspend">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>225</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>挂起队列</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTableView" name="tableEnd">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>50</y>
      <width>320</width>
      <height>239</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>320</width>
      <height>520</height>
     </size>
    </property>
   </widget>
   <widget class="QTableView" name="memTable">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>50</y>
      <width>230</width>
      <height>239</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>230</width>
      <height>0</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>230</width>
      <height>520</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="button_beginScheduling">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>600</y>
      <width>110</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>自动调度</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_stopScheduling">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>600</y>
      <width>110</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>暂停调度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_suspend_2">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>295</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>添加进程</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar_memuse">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>610</y>
      <width>124</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>90</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>610</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>内存使用率：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>650</y>
      <width>101</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>内存空闲率：</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar_free">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>650</y>
      <width>124</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QPushButton" name="button_randomadd">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>620</y>
      <width>141</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>随机生成5个进程</string>
    </property>
   </widget>
   <widget class="QSlider" name="speed_control">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>630</y>
      <width>160</width>
      <height>22</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>600</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>执行速度：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_suspend_3">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>300</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>前驱后继图</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_suspend_4">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>395</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>阻塞队列</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QTableView" name="tableblock">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>430</y>
      <width>295</width>
      <height>131</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>295</width>
      <height>400</height>
     </size>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1303</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuOS_Master_JsingMog">
    <property name="title">
     <string>菜单</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionSuspend"/>
    <addaction name="separator"/>
    <addaction name="actionUnsuspend"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>设置</string>
    </property>
    <addaction name="action_setcpunum"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuOS_Master_JsingMog"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSuspend">
   <property name="text">
    <string>进程挂起
</string>
   </property>
  </action>
  <action name="actionUnsuspend">
   <property name="text">
    <string>进程解挂</string>
   </property>
  </action>
  <action name="action_setcpunum">
   <property name="text">
    <string>处理机数量
</string>
   </property>
  </action>
  <action name="action_times">
   <property name="text">
    <string>时间片大小</string>
   </property>
  </action>
  <action name="action_mainmem">
   <property name="text">
    <string>主存空间大小</string>
   </property>
  </action>
  <action name="actionIO">
   <property name="text">
    <string>进程IO</string>
   </property>
  </action>
  <action name="actionIOend">
   <property name="text">
    <string>IO结束</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
